File: src/test/programmers/java/level2/RepeatBinaryTransformation.java
package test.programmers.java.level2;

import java.util.Arrays;
import java.util.stream.Collectors;

public class RepeatBinaryTransformation {
    public static void main(String[] args) {
        System.out.println(Arrays.toString(solution("01110")));
    }

    static public int[] solution(String s) {
        int[] answer = {};
        int count = 0;
        int zeroCount = 0;

        while(!s.equals("1")) {
            String[] str = s.split("");
            zeroCount += Arrays.stream(str).filter(i -> i.equals("0")).count();
            int length = Arrays.stream(str).filter(i -> i.equals("1")).collect(Collectors.joining()).length();
            s = Integer.toBinaryString(length);
            count++;
        }

        answer = new int[]{count, zeroCount};
        return answer;
    }
}

File: src/test/programmers/python/level2/RepeatBinaryTransformation.py
def solution(s):
    zero_count, try_count = 0, 0
    while len(s) != 1:
        zero_count += s.count('0')
        s = s.replace('0', '')
        try_count += 1
        s = bin(len(s))[2:]

    return [try_count, zero_count]

print(solution("110010101001"))